al_aws.js

```diff
@@ -34,6 +36,132 @@ var selfUpdate = function (callback) {
 };
 
 
+var selfConfigUpdate = function (callback) {
+    async.waterfall([
+        function(callback) {
+            getConfigChanges(function(err, config) {
+                callback(err, config)
+            });
+        },
+        function(newValues, callback) {
+            getCurrentConfig(function(err, currentConfig) {
+                callback(err, newValues, currentConfig)
+            });
+        },
+        function(newValues, currentConfig, callback) {
+            applyConfigChanges(newValues, currentConfig, function(err, newConfig) {
+                callback(err, newConfig, currentConfig)
+            })
+        },
+        function(newConfig, currentConfig, callback) {
+            if (isConfigDifferent(newConfig, currentConfig)) {
+                var lambda = new AWS.Lambda();
+                var updateConfig = filterDisallowedConfigParams(newConfig);
+                return lambda.updateFunctionConfiguration(updateConfig, callback);
+            } 
+            else {
+                callback(null);
+            }
+        }
+    ],
+    function(err, config) {
+        if (err) {
+            console.info('Lambda self-update config error: ', err);
+        } else {
+            if (config !== undefined) {
+                console.info('Lambda self-update config successful. Config: ', config);
+            }
+            else {
+                console.info('Lambda self-update config nothing to update');
+            }
+        }
+        callback(err, config);
+    });
+};
+
+function getConfigChanges(callback) {
+    var s3 = new AWS.S3();
+    var configName;
+    
+    // Use this default config in order to update old collectors
+    if (process.env.aws_lambda_update_config_name == undefined) {
+        configName = DEFAULT_CONFIG_NAME;
+    }
+    else {
+        configName = process.env.aws_lambda_update_config_name 
+    }
+    
+    var params = {
+        Bucket: process.env.aws_lambda_s3_bucket,
+        Key: configName
+    };
+    s3.getObject(params, function(err, object) {
+        if (err) {
+            return callback(err);
+        } else {
+            let config = JSON.parse(object.Body.toString());
+            return callback(null, config);
+        }
+    });
+}
+
+function getCurrentConfig(callback) {
+    var lambda = new AWS.Lambda();
+    var params = {
+        FunctionName: process.env.AWS_LAMBDA_FUNCTION_NAME
+    };
+    
+    lambda.getFunctionConfiguration(params, callback);
+}
+
+function applyConfigChanges(newValues, config, callback) {
```

Comment:
I'd suggest to move config changes calculation to `AlAwsCollector` class. And leave AWS API wrappers, like `getLambdaConfigS3()` here.

Reply:
I am quite reluctant to do it as I don't see any benefits and it would mean to refactor all tests significantly because of the class. Everything in the module is related to the config update and there is no benefit to have it in the class.

Reply:
The goal is to have `AlAwsCollector.update()` related logic incapsulated into the class, while it is ok to keep AWS API wrappers here. Could you please move it in oder not to create tech debt?

Ideally we need to refactor `selfUpdate()` and pass update related params from `AlAwsCollector` class.